apps/web/src/app/[locale]/tracks/_components/track-details.tsx

```diff
@@ -50,6 +51,18 @@ export async function TrackDetail({ slug }: TrackDetailProps) {
     })
     .map((trackChallenge) => trackChallenge.id);
 
+  async function enroll(id: number) {
```

Comment:
can we just move the track to the server actions that are defined already

Reply:
Ya I was initially doing it that way only but it was resulting in code changes in multiple places, below is the updated commit.